src/Map.cpp

```diff
@@ -144,4 +144,28 @@ void Map::printNodes() {
         std::cout << std::endl;
         nodeIter++;
     }
+}
+
+int Map::getNodeIndex(std::string regionName) {
+  auto nodeIter = nodes->begin();
+  int result = -1;
+  int index = 0;
+  while (nodeIter != nodes->end()) {
+    if (nodeIter->region == regionName) {
+      result = index;
+      break;
+    }
+    index++;
+    nodeIter++;
+  }
+}
+
+bool Map::addArmy(std::string regionName, std::string playerName) {
+  int nodeIndex = getNodeIndex(regionName);
+  if (nodeIndex == -1) {
+    return false;
+  }
```

Comment:
Thinking about this now, we're missing a check on if the playername is valid. Would it be possible to add this check in any calls in `Player`? If not I can keep a list of player names in `Map` and validate in there. I'm good either way.

Reply:
playerName should always be valid. When we constructor the Players vector at the start of the game we will only accept valid strings and then that is what will be passed for the remainder of the game.